frontend/src/component/project/Project/ProjectSettings/ProjectActions/ProjectActionsTable/ProjectActionsModal/ProjectActionsForm/ProjectActionsFormStep/ProjectActionsFormStepActions/useActionDefinitions.ts

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from 'react';
+import { ActionDefinitionParameter } from '@server/util/constants/action-parameters';
+import { ACTIONS, ActionDefinition } from '@server/util/constants/actions';
+import { useFeatureSearch } from 'hooks/api/getters/useFeatureSearch/useFeatureSearch';
+import useProjectOverview from 'hooks/api/getters/useProjectOverview/useProjectOverview';
+
+type ActionDefinitionParameterWithOption = ActionDefinitionParameter & {
+    options: string[];
+};
+
+type ActionDefinitionWithParameterOptions = Omit<
+    ActionDefinition,
+    'parameters'
+> & {
+    parameters: ActionDefinitionParameterWithOption[];
+};
+
+export type ActionDefinitions = Map<
+    string,
+    ActionDefinitionWithParameterOptions
+>;
+
+export const useActionDefinitions = (projectId: string): ActionDefinitions => {
+    const { project, loading: isProjectLoading } =
+        useProjectOverview(projectId);
+    const { features, loading: isFeaturesLoading } = useFeatureSearch({
+        project: `IS:${projectId}`,
+    });
+
+    const [actionDefinitions, setActionDefinitions] =
+        useState<ActionDefinitions>(new Map());
+
+    useEffect(() => {
+        if (isProjectLoading || isFeaturesLoading) return;
+
+        const optionsByType: Record<
+            ActionDefinitionParameter['type'],
+            string[]
+        > = {
+            project: [],
+            environment: project.environments.map(
+                ({ environment }) => environment,
+            ),
+            featureToggle: features.map(({ name }) => name).sort(),
+        };
+
+        const actionDefinitionsWithParameterOptions = new Map<
+            string,
+            ActionDefinitionWithParameterOptions
+        >(
+            [...ACTIONS].map(([key, action]) => [
```

Comment:
Should we use here actionDefinitions?

Reply:
I believe this is correct since here we want to _extend_ the base definition of actions (from the shared constant in `@server/util/constants/actions`) to include the options dynamically. `actionDefinitions` will be our end result.